src/ape/managers/chain.py

```diff
@@ -0,0 +1,424 @@
+        ``-2`` would be the block prior to that one, and so on.
+
+        Args:
+            block_number (int): The number of the block to get.
+
+        Returns:
+            :class:`~ape.api.providers.BlockAPI`
+        """
+
+        if block_number < 0:
+            block_number = len(self) + block_number
+
+        return self._get_block(block_number)
+
+    def __len__(self) -> int:
+        """
+        The number of blocks in the chain.
+
+        Returns:
+            int
+        """
+
+        return self.height + 1
+
+    def __iter__(self) -> Iterator[BlockAPI]:
+        """
+        Iterate over all the current blocks.
+
+        Returns:
+            Iterator[:class:`~ape.api.providers.BlockAPI`]
+        """
+
+        return self.range()
+
+    def range(self, start: int = 0, stop: Optional[int] = None) -> Iterator[BlockAPI]:
+        """
+        Iterate over blocks. Works similarly to python ``range()``.
+
+        Args:
+            start (int): The first block, by number, to include in the range.
+              Defaults to 0.
+            stop (Optional[int]): The block number to stop before. Also the total
+             number of blocks to get.
+
+        Returns:
+            Iterator[:class:`~ape.api.providers.BlockAPI`]
+        """
+
+        if stop is None:
+            stop = len(self)
+
+        if stop > len(self):
+            raise ChainError(
+                f"'stop={stop}' cannot be greater than the chain length ({len(self)}). "
+                f"Use '{self.poll_blocks.__name__}()' to wait for future blocks."
+            )
+        elif stop < start:
+            raise ValueError(f"stop '{stop}' cannot be less than start '{start}'.")
+        elif stop < 0:
+            raise ValueError(f"start '{start}' cannot be negative.")
+        elif start < 0:
+            raise ValueError(f"stop '{stop}' cannot be negative.")
+
+        for i in range(start, stop):
+            yield self._get_block(i)
+
+    def poll_blocks(
+        self,
+        start: Optional[int] = None,
+        required_confirmations: Optional[int] = None,
+        poll_wait_interval: int = 2,
+    ) -> Iterator[BlockAPI]:
+        """
+        Poll new blocks. Optionally set a start block to include historical blocks.
+        **NOTE**: This is a deamon method; it does not terminate unless an exception
+        occurrs.
+
+        Usage example::
+
+            from ape import chain
+
+            for new_block in chain.blocks.poll_blocks():
+                print(f"New block found: number={new_block.number}")
+
+        Args:
+            start (Optional[int]): The block number to start with. Defaults to the block
+              of the ``+1`` the current block number.
+            required_confirmations (Optional[int]): The amount of confirmations to wait
+              before yielding the block.
+            poll_wait_interval (int): The amount of seconds to wait before checking for
+              new blocks. Defaults to ``2``.
```

Comment:
can probably default to a portion of expected block time (which is different chain!)

Reply:
Oh that is smart